### PR TITLE
Editor load video

### PR DESCRIPTION
**Pull request by Nolski**
This updates the video editor page to now auto load the video into the editor.

This PR includes changes to popcorn-editor.js as well. To mess around the UI changes head to http://{domain}/popcorn/edit/{slug}

### CODE REVIEW

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 1]
```diff
@@ -1,2 +1,27 @@
...
+
+PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+  $.ajax({
+    method: 'GET',
+    url: '/popcorn/data/',
+    contentType: 'application/json',
+    data: {
+      slug: getSlug()
+    }
+  })
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: You can actually just do
```javascript
$.get('/popcorn/data/', {slug: getSlug()})
.done...
```

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+      re = /\/popcorn\/edit\/(.+)\//,
+      match = pathname.match(re);
+
+  return match[1];
+}
+
+PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+  $.ajax({
+    method: 'GET',
+    url: '/popcorn/data/',
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: We need to eradicate all hardcoded URLs from the static files. 
This should only exist has a written string in one single place `popcorn/urls.py`.
The way to "transfer" such a value from django into a script is to use a tag and a `data` attribute.  

Here's one example: https://github.com/mozilla/airmozilla/blob/master/airmozilla/main/templates/main/main_base.html#L169-L175
https://github.com/mozilla/airmozilla/blob/master/airmozilla/main/static/main/js/autocompeter.js#L7-L11

You don't have to use a non-standard tag. Perhaps you can attach...
```
data-editor-url="{{ url('popcorn:data') }}"
```
... on the `body` tag or something.

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 3]
```diff
@@ -1,2 +1,27 @@
...
+      slug: getSlug()
+    }
+  })
+  .done(function (response) {
+    if(response.exists) {
+      PopcornEditor.loadInfo(response.data);
+    } else {
+      PopcornEditor.createTemplate(response.data);
+    }
+  });
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: This lacks a failure resolver. If you can't think of one at least start with something like this:
```javascript
.fail(function() {
    console.warn("Unable to load popcorn data :(");
    console.error.apply(console, arguments);
})
```
Or, if you don't have time, add an `alert()` because otherwise only people who open the web console will be informed.

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 4]
```diff
@@ -1,2 +1,27 @@
...
+    contentType: 'application/json',
+    data: {
+      slug: getSlug()
+    }
+  })
+  .done(function (response) {
+    if(response.exists) {
+      PopcornEditor.loadInfo(response.data);
+    } else {
+      PopcornEditor.createTemplate(response.data);
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: This doesn't make sense any more after what we decided.

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 5]
```diff
@@ -1,2 +1,27 @@
+function getSlug() {
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: This whole file needs to be wrapped in something like `$(function() { ... })` so no javascript magic happens until everything has been loaded. 

And ideally I would use a closure so you don't add `getSlug()` to the `window` global scope. 

You can do this pattern if you like:
```
var PopcornIntegration = (function() {
    function getSlug() {
    }
    return {
        init: function() {
          PopcornEditor.listen(...)
        }
    }
});

$(function() {
    PopcornIntegration.init();
});
```

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 6]
```diff
@@ -1,2 +1,27 @@
...
+  var pathname = window.location.pathname,
+      re = /\/popcorn\/edit\/(.+)\//,
+      match = pathname.match(re);
+
+  return match[1];
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: This whole function ought to be refactored so it doesn't depend on the URL. 
Where I said to use a `data` attribute to transfer the value of the URL from Django to the javascript via a django template, do the same with the value of the slug.

File: airmozilla/popcorn/views.py [hunk 7]
```diff
@@ -51,11 +42,72 @@ def event_meta_data(request):
...
+    slug = request.GET.get('slug')
+    event = get_object_or_404(Event, slug=slug)
+    pattern = re.compile('(.+)\?')
+
+    base_url = get_video_url(event)
+    matches = pattern.match(base_url)
+    if matches:
+        video_url = matches.group()[:-1]
+    else:
+        video_url = base_url
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: Don't use a regex. Use `urlparse.urlparse` and `urlparse.urljoin` to drop the randomish cache-buster thingy.

File: airmozilla/popcorn/views.py [hunk 8]
```diff
@@ -51,11 +42,72 @@ def event_meta_data(request):
...
+        break
+    else:
+        video_url = get_video_url(event)
+        thumb = get_thumbnail(event)
+        data = {
+            "thumbnail": thumb.url,
+            "url": video_url,
+            "title": event.title,
+            "duration": event.duration,
+            "type": "AirMozilla",
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: What is this `type`? If it's something that ever reaches users' eyes it should be `Air Mozilla`.

Comment (Nolski) [inline airmozilla/popcorn/views.py]: This is used within Popcorn Editor and changes how it loads the video data. It refers to the [regex map](https://github.com/Nolski/popcorn-editor/blob/master/PopcornEditor/src/util/mediatypes.js#L9) for loading in different media types

File: airmozilla/popcorn/views.py [hunk 9]
```diff
@@ -51,11 +42,72 @@ def event_meta_data(request):
...
+        }
+
+        return {
+            "exists": False,
+            "data": data,
+        }
+
+    return {
+        "exists": True,
+        "data": data,
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: DRY. 

As discussed in IRC, use `response.data` or `response.metadata` as a boolean expression in the ajax callback.

**Comment by peterbe** [comment]
It's great! We just need to fix a couple of nits. In particular to get away from relying on hardcoded URLs and `document.location`

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/PopcornEditor.js [hunk 10]
```diff
@@ -44,6 +44,85 @@ var PopcornEditor = (function () {
...
+                        "thumbnail": video.thumbnail,
+                        "duration": video.duration
+                    }
+                },
+                "currentTime": 0,
+            }]
+        },
+        "tags": ["popcorn"],
+    }
+    this.loadInfo(data);
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/PopcornEditor.js]: The function implies that it creates stuff but it actually does a load. Perhaps refactor to
`PopcornEditor.loadInfo(PopcornEditor.createTemplate(response.video_metadata))`

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js [hunk 11]
```diff
@@ -284,11 +286,11 @@ define( [ "util/lang", "util/uri", "util/xhr", "util/keys", "util/mediatypes", "
     resetInput();
   }
 
-  function onSuccess( data ) {
+  function onSuccess( data, imported ) {
     var el = _GALLERYITEM.cloneNode( true ),
         source = data.source;
 
-    if ( !_media.clipData[ source ] ) {
+    if ( !_media.clipData[ source ]  || imported !== undefined) {
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js]: Why does this matter?
If there's a good reason, leave a comment that explains the necessity to distinguish between imported and non-imported.

File: airmozilla/popcorn/static/popcorn/js/editor.js (hunk 4 above)
Comment (Nolski) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: Had a chat offline about this. Just a little misunderstanding.

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 12]
```diff
@@ -1,2 +1,30 @@
...
+    var config = $('slug');
+    if (config.length) {
+      return config.data('url');
+    } else {
+      return '';
+    }
+  }
+
+  PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+    console.log('slug', getSlug());
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: debugging?

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 13]
```diff
@@ -1,2 +1,30 @@
-// Initialize the editor with the div id and path to Popcorn Editor.
-PopcornEditor.init('editor', '/static/popcorn/PopcornEditor/editor.html');
+$(document).ready(function () {
+  function getSlug() {
+    var config = $('slug');
+    if (config.length) {
+      return config.data('url');
+    } else {
+      return '';
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: Why would this ever happen?

File: airmozilla/popcorn/templates/popcorn/editor.html [hunk 14]
```diff
@@ -23,6 +23,7 @@
 {% endblock %}
 
 {% block full_width_content %}
+<slug data-url="{{ slug }}"></slug>
 <div id="editor"></div>
```
Comment (peterbe) [inline airmozilla/popcorn/templates/popcorn/editor.html]: You could put that `data-url={}` thing on the `<div id="editor">` element which you already have.

File: airmozilla/popcorn/tests/test_views.py [hunk 15]
```diff
@@ -54,3 +54,42 @@ def mocked_head(url, **options):
...
+        # because we're not logged in
+        eq_(response.status_code, 302)
+
+        self._login()
+
+        response = self.client.get(url, {'slug': event.slug})
+        eq_(response.status_code, 200)
+
+        content = json.loads(response.content)
+        print content
```
Comment (peterbe) [inline airmozilla/popcorn/tests/test_views.py]: debugging

File: airmozilla/popcorn/views.py [hunk 16]
```diff
@@ -1,5 +1,6 @@
-import requests
 from jsonview.decorators import json_view
+import requests
+from urlparse import urlparse
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: `urlparse` is a standard python module and its import line should go first and separated (by a newline) from the third-party modules.

File: airmozilla/popcorn/views.py [hunk 17]
```diff
@@ -51,11 +42,68 @@ def event_meta_data(request):
 
     assert response.status_code == 302, response.status_code
 
+    purl = urlparse(response.headers['location'])
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: Comment missing that explains *why* you're doing this. It's because of the `?t=091u23490-123` thing that vid.ly adds, right? And the reason for doing that is because it prevents popcorn-editor to index the media by its video URL.

File: airmozilla/popcorn/views.py [hunk 18]
```diff
@@ -51,11 +42,68 @@ def event_meta_data(request):
...
+    crop = 'center'
+
+    return thumbnail(image, geometry, crop=crop)
+
+
+@add_cors_header('*')
+@json_view
+def event_meta_data(request):
+    slug = request.GET.get('slug')
+    event = get_object_or_404(Event, slug=slug)
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: You're using a `.get()`, which implies that you're OK with that it might become `slug=None`. Then you can't do the next line. The `get_object_or_404`. 

I would change these two lines to:
```python
if not request.GET.get('slug'):
    return http.HttpResponseBadRequest('slug')
event = get_object_or_404(Event, slug=request.GET['slug'])
```

Comment (peterbe) [inline airmozilla/popcorn/views.py]: One cool feature in tornado (the event driven python web framework) is that you pick things up from the request using `self.get_argument('slug')` and it automatically raises a 400 error for you if that variable is missing or an empty string. And if you don't want that to be required you can do things like: `order_by = self.get_argument('order', 'created')` which sets a default.

File: airmozilla/popcorn/views.py [hunk 19]
```diff
@@ -51,11 +42,68 @@ def event_meta_data(request):
...
+
+    video_url = get_video_url(event)
+
+    thumb = get_thumbnail(event)
+
     return {
         'title': event.title,
         'description': event.short_description or event.description,
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: Careful as this might contain HTML tags. There's a helper function somewhere called `unhtml` which you might be interested in.

Comment (peterbe) [inline airmozilla/popcorn/views.py]: Also, the description can be *very* long. How does Popcorn editor cope with that?

File: airmozilla/popcorn/views.py [hunk 20]
```diff
@@ -51,11 +42,68 @@ def event_meta_data(request):
...
-        'video_url': response.headers['location'],
+        'video_url': video_url,
+    }
+
+
+@json_view
+@login_required
+def popcorn_data(request):
+    slug = request.GET.get('slug')
+    event = get_object_or_404(Event, slug=slug)
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: ditto

File: airmozilla/popcorn/views.py [hunk 21]
```diff
@@ -51,11 +42,68 @@ def event_meta_data(request):
...
+            "thumbnail": thumb.url,
+            "url": video_url,
+            "title": event.title,
+            "duration": event.duration,
+            "type": "AirMozilla",
+        }
+
+        return {
+            "metadata": data,
+            "slug": slug,
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: Why return what you already clearly knew when you called this view?

File: airmozilla/popcorn/views.py [hunk 22]
```diff
@@ -92,7 +139,8 @@ def get(self, request, slug, form=None, conflict_errors=None):
             return self.cant_edit_event(event, request)
 
         context = {
-            'event': event
+            'event': event,
+            'slug': slug,
```
Comment (peterbe) [inline airmozilla/popcorn/views.py]: ditto.

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js (hunk 11 above)
Comment (Nolski) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js]: This prevents you from loading in a video (through the import media dialogue) when it was already loaded in through the API

I'll leave a comment

File: airmozilla/popcorn/templates/popcorn/editor.html (hunk 14 above)
Comment (Nolski) [inline airmozilla/popcorn/templates/popcorn/editor.html]: Wouldn't it be a bit more semantic to have the slug be a part of the whole page though? It isn't necessarily specific to the editor element.

File: airmozilla/popcorn/views.py (hunk 19 above)
Comment (Nolski) [inline airmozilla/popcorn/views.py]: It will just ellipse after a certain amount of characters. HTML tags should be handled by the js template engine

File: airmozilla/popcorn/views.py (hunk 22 above)
Comment (Nolski) [inline airmozilla/popcorn/views.py]: The slug needs to exist within the HTML for the javascript to access which requires it to exist within the context.

File: airmozilla/popcorn/templates/popcorn/editor.html (hunk 14 above)
Comment (peterbe) [inline airmozilla/popcorn/templates/popcorn/editor.html]: Yes and no. It's a matter of foresight. Like with a global `styles.css` you *could* add a selector there that is only needed on one page. But there's a big chance you decommission that page in the far future but forget to delete it from `styles.css` and then you have a bloated `styles.css` that future developers get uneasy about cleaning up. That's why I favor using `<style>` tags on individual pages that only need a specific set of rules. (If it gets too big I wrote a thing like `eventedit.css` instead of an inline tag in the HTML)

Then, if I find a need a thing on more than just one page. I can either reuse `eventedit.css` on multiple pages. Or if it gets too many pages, it gets promoted to be included in the global `styles.css`. 

Does that help to explain the issue?

Basically, what if we change how cornpop.js works and nothing no longer needs that `data-slug="..."` in `main_base.html` then `main_base.html` gets scarily bloated.

File: airmozilla/popcorn/views.py (hunk 22 above)
Comment (peterbe) [inline airmozilla/popcorn/views.py]: Seems strange but if you say so, let's leave it then. 
"Hey Mike, what's your first name? Mike? Oh cool. Thanks Mike for informing me"  :)

**Comment by Nolski** [comment]
@peterbe once tests pass this should be good to look at again.

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/PopcornEditor.js [hunk 23]
```diff
@@ -44,6 +44,85 @@ var PopcornEditor = (function () {
...
+                        "thumbnail": video.thumbnail,
+                        "duration": video.duration
+                    }
+                },
+                "currentTime": 0,
+            }]
+        },
+        "tags": ["popcorn"],
+    }
+    return data;
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/PopcornEditor.js]: Yay!

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js [hunk 24]
```diff
@@ -700,6 +702,16 @@ define( [ "util/lang", "util/uri", "util/xhr", "util/keys", "util/mediatypes", "
     _searchInput.addEventListener( "keydown", onEnter );
 
     _addBtn.addEventListener( "click", onAddMediaClick );
+
+    Event.listen('load', function (data) {
+      var clipData = data.data.media[0].clipData,
+          clip;
+
+      for (url in clipData) {
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js]: These days you can let yourself use the `clipData.forEach(function(clip) {...` pattern.

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 25]
```diff
@@ -1,2 +1,29 @@
-// Initialize the editor with the div id and path to Popcorn Editor.
-PopcornEditor.init('editor', '/static/popcorn/PopcornEditor/editor.html');
+$(document).ready(function () {
+  function getSlug() {
+    var config = $('slug');
+    if (config.length) {
+      return config.data('url');
+    } else {
+      console.error('Slug not provided in DOM. Cannot get slug');
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: I would do `throw 'Slug not provided...'`

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 26]
```diff
@@ -1,2 +1,29 @@
...
+    }
+  }
+
+  PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+    $.get('/popcorn/data/', {
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: This too ought to come from a `data-` tag somewhere. No hardcoded URLs please. URLs should only be "hardcoded" in one place and that's `urls.py`

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js (hunk 24 above)
Comment (Nolski) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js]: clipData is an `object` not an `Array`

File: airmozilla/popcorn/static/popcorn/js/editor.js (hunk 26 above)
Comment (Nolski) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: I'm not quite sure how this would be kept in `urls.py` the path to the editor.html file is _strictly_ related to the PopcornEditor.js code. I could hard code it into the template file but that also seems to disconnect it from the actual code that it is semantically linked to.

Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: So that URL there relates to [this](https://github.com/Nolski/airmozilla/blob/editor-load-video/airmozilla/popcorn/urls.py#L13-L15) right?
Once you have a URL written down in a `urls.py` you're supposed to refer to it by the name. It's worth it in the long run. 

This line would be replaced with `$('slug').data('data-url')` or something. 
And in the template you'd do something like this:
```html
<slug data-data-url="{{ url('popcorn:data') }}">
```

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js (hunk 24 above)
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js]: dang it! Good catch. Perhaps put in a comment that explains that for the next person who has the same thought as me.

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 27]
```diff
@@ -1,2 +1,29 @@
...
+    .done(function (response) {
+      if(response.data) {
+        PopcornEditor.loadInfo(response.data);
+      } else {
+        PopcornEditor.loadInfo(PopcornEditor.createTemplate(response.metadata));
+      }
+    })
+    .fail(function() {
+      console.warn("Unable to load popcorn data :(");
+      console.error.apply(console, arguments);
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: These actually don't help the user that much. Only if they know how to enable the web developer web console will the get a hint that if finished loading but didn't work. 

Couldn't you do something like 
```javascript
.fail(function() {
    ...same console.error lines as before...
    $('#load-error').show();
})
```
And in `editor.html` something like
```html
<div id="load-error" style="display:none">
<p>Someone something something</p>
</div>
```

File: airmozilla/popcorn/templates/popcorn/editor.html [hunk 28]
```diff
@@ -23,6 +23,7 @@
 {% endblock %}
 
 {% block full_width_content %}
+<slug data-url="{{ slug }}"></slug>
```
Comment (peterbe) [inline airmozilla/popcorn/templates/popcorn/editor.html]: I'm still not convinced we need this extra tag. We should be able to put this into the `<div id="editor">` tag.

Comment (peterbe) [inline airmozilla/popcorn/templates/popcorn/editor.html]: That means, just being part of the `<div id="editor">` tag it means that those data blocks are related to the editor thing.

File: airmozilla/popcorn/static/popcorn/js/editor.js (hunk 26 above)
Comment (Nolski) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: Nope that url refers to [this](https://github.com/Nolski/airmozilla/blob/editor-load-video/airmozilla/popcorn/static/popcorn/PopcornEditor/editor.html) which is a static HTML file that belongs to the PopcornEditor

File: airmozilla/popcorn/static/popcorn/js/editor.js (hunk 27 above)
Comment (Nolski) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: This is on my TODO list but seeing as exporting edits also isn't yet implemented either I kind of put that on the back burner.

File: airmozilla/popcorn/templates/popcorn/editor.html (hunk 28 above)
Comment (Nolski) [inline airmozilla/popcorn/templates/popcorn/editor.html]: fair enough

File: airmozilla/popcorn/static/popcorn/js/editor.js (hunk 27 above)
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: fair enough.

File: airmozilla/popcorn/static/popcorn/js/editor.js (hunk 26 above)
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: Isn't `/popcorn/data/` a URL in `urls.py` in django?

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 29]
```diff
@@ -1,2 +1,29 @@
-// Initialize the editor with the div id and path to Popcorn Editor.
-PopcornEditor.init('editor', '/static/popcorn/PopcornEditor/editor.html');
+$(document).ready(function () {
+  function getSlug() {
+    var config = $('slug');
+    if (config.length) {
+      return config.data('url');
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: When you remove `<slug>` and use `<div id="editor">` instead, change the name from `url` to `slug`.

**Comment by peterbe** [comment]
Ping me in IRC so we can figure out why your schemamigrations stuff doesn't want to generate a new migration file.

**Comment by Nolski** [comment]
@peterbe Looks like I had to drop the test table to get tests to pass (also needed to add the database migration)

File: airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js (hunk 11 above)
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/PopcornEditor/src/editor/media-gallery-editor.js]: Don't forget that comment. I think it's an important underlying design thing to understand.

File: airmozilla/popcorn/static/popcorn/js/editor.js [hunk 30]
```diff
@@ -1,2 +1,20 @@
-// Initialize the editor with the div id and path to Popcorn Editor.
-PopcornEditor.init('editor', '/static/popcorn/PopcornEditor/editor.html');
+$(document).ready(function () {
+  PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+    $.get($('#editor').data('url'), {
+        slug: $('#editor').data('slug')
+    })
```
Comment (peterbe) [inline airmozilla/popcorn/static/popcorn/js/editor.js]: Awesome! Keeps it simple.

File: airmozilla/popcorn/templates/popcorn/editor.html [hunk 31]
```diff
@@ -23,7 +23,7 @@
 {% endblock %}
 
 {% block full_width_content %}
-<div id="editor"></div>
+<div id="editor" data-slug="{{ slug }}" data-url="{{ url(popcorn:data) }}"></div>
```
Comment (peterbe) [inline airmozilla/popcorn/templates/popcorn/editor.html]: How could this have worked in testing? Perhaps we forgot to add a test that executes this view. 
It's supposed to be `{{ url('popcorn:data') }}`. 

Check your coverage first and then, write a test that executes this template (which should fail because the expression `{{ url(popcorn:data) }}` is most likely not valid in Jinja.

File: airmozilla/popcorn/tests/test_views.py [hunk 32]
```diff
@@ -54,3 +54,41 @@ def mocked_head(url, **options):
...
+        event = Event.objects.get(title='Test event')
+        event.template.name = 'this is a vid.ly video'
+        event.template.save()
+
+        event.template_environment = {'tag': 'abc123'}
+        event.save()
+
+        url = reverse('popcorn:popcorn_data')
+
+        response = self.client.get(url, {'slug': event.slug})
```
Comment (peterbe) [inline airmozilla/popcorn/tests/test_views.py]: Before this do...
```python
response = self.client.get(url)
# bad request because it's missing the slug
eq_(response.status_code, 400)

response = self.client.get(url, {'slug': ''})
# still not a valid slug value
eq_(response.status_code, 400)
```

**Comment by peterbe** [comment]
Seems to lack test coverage in a couple of places. 
The rendering of the page that has `<div id="editor" data-slug="{{ slug }}">` might deserve its own test. 

Also you have no test coverage of when there is 1 or more PopcornEdits. E.g. that for-loop with the break in it. 

If you can't remember how the coverage was done, see [the readme](https://github.com/mozilla/airmozilla#tests-and-test-coverage).

I'm really eager to see this land. We are so close! I'll help by building a little admin thingy so you can get insights into known edits.

**Comment by Nolski** [comment]
@peterbe squashed and rebased.
